rust/otap-dataflow/crates/otap/src/otap_batch_processor.rs

```diff
@@ -0,0 +1,698 @@
+    fn default() -> Self {
+        Self {
+            send_batch_size: default_send_batch_size(),
+            send_batch_max_size: default_send_batch_max_size(),
+            timeout: default_timeout_ms(),
+            metadata_keys: Vec::new(),
+            metadata_cardinality_limit: None,
+        }
+    }
+}
+
+/// Local (!Send) OTAP batch processor
+pub struct OtapBatchProcessor {
+    config: Config,
+    /// Single buffer of incoming pdata messages. We partition by signal type at flush.
+    current: Vec<OtapPdata>,
+}
+
+impl OtapBatchProcessor {
+    /// Construct a processor wrapper from a JSON configuration object and processor runtime config.
+    /// The JSON should mirror the Go collector batchprocessor shape. Missing fields fall back to
+    /// crate defaults. Invalid numeric values (e.g., zero) are normalized to minimal valid values.
+    pub fn from_config(
+        node: NodeId,
+        cfg: &Value,
+        proc_cfg: &ProcessorConfig,
+    ) -> Result<ProcessorWrapper<OtapPdata>, ConfigError> {
+        let mut config: Config = serde_json::from_value(cfg.clone()).unwrap_or_default();
+
+        // Accept Go-style duration strings for timeout (e.g., "200ms", "2s", "1m", "1m30s").
+        // If provided as a string, parse like Go's time.ParseDuration; if numeric, keep as ms.
+        if let Some(timeout_val) = cfg.get("timeout") {
+            if let Some(s) = timeout_val.as_str() {
+                if let Some(ms) = parse_duration_ms(s) {
+                    config.timeout = ms;
+                }
+            }
+        }
+
+        // Basic validation/normalization
+        if config.send_batch_size == 0 {
+            config.send_batch_size = MIN_SEND_BATCH_SIZE;
+        }
+        // Go behavior: if send_batch_max_size is 0 (sentinel), use send_batch_size
+        let effective_sbs = config.send_batch_size;
+        let max = if config.send_batch_max_size == FOLLOW_SEND_BATCH_SIZE_SENTINEL {
+            effective_sbs
+        } else {
+            config.send_batch_max_size
+        };
+        config.send_batch_max_size = max;
+
+        if let Some(limit) = config.metadata_cardinality_limit {
+            if limit < MIN_METADATA_CARDINALITY_LIMIT {
+                config.metadata_cardinality_limit = Some(MIN_METADATA_CARDINALITY_LIMIT);
+            }
+        }
+        let user_config = Arc::new(NodeUserConfig::new_processor_config(
+            OTAP_BATCH_PROCESSOR_URN,
+        ));
+        let proc = OtapBatchProcessor {
+            config,
+            current: Vec::new(),
+        };
+        Ok(ProcessorWrapper::local(proc, node, user_config, proc_cfg))
+    }
+
+    /// Returns true if the buffered items reach or exceed the emission threshold.
+    ///
+    /// MVP: threshold is based on count (number of messages). In future, switch to real
+    /// row counts for OtapArrowRecords.
+    fn size_triggers_emission(&self) -> bool {
+        self.current.len() >= self.config.send_batch_size
+    }
```

Comment:
I agree, long term threshold should use the total size of the batches instead of the count of batches. We should be able to do this now because in #994 a method was added to calculate the current batch size. This is what we should be using. 
https://github.com/open-telemetry/otel-arrow/blob/8c4473eefcd9f1107e49a56bb26d3b74d4c97ab0/rust/otel-arrow-rust/src/otap.rs#L96-L101

e.g. to get the size of the pdata, we convert it to OTAP then and then call this method:
```
// probably best to do this in `process`:
let otap: OtapArrowRecords = pdata.try_into()?;
let batch_len = otap.batch.length();
```

Reply:
I have 3 current vecs for each telemetry type now along wuith seperate counts for each.